pkg/image/apiserver/apiserver.go

```diff
@@ -0,0 +1,193 @@
+func (c *ImageAPIServerConfig) Complete() completedConfig {
+	c.GenericConfig.Complete()
+
+	return completedConfig{c}
+}
+
+// SkipComplete provides a way to construct a server instance without config completion.
+func (c *ImageAPIServerConfig) SkipComplete() completedConfig {
+	return completedConfig{c}
+}
+
+// New returns a new instance of ImageAPIServer from the given config.
+func (c completedConfig) New(delegationTarget genericapiserver.DelegationTarget) (*ImageAPIServer, error) {
+	genericServer, err := c.ImageAPIServerConfig.GenericConfig.SkipComplete().New("image.openshift.io-apiserver", delegationTarget) // completion is done in Complete, no need for a second time
+	if err != nil {
+		return nil, err
+	}
+
+	s := &ImageAPIServer{
+		GenericAPIServer: genericServer,
+	}
+
+	v1Storage, err := c.V1RESTStorage()
+	if err != nil {
+		return nil, err
+	}
+
+	apiGroupInfo := genericapiserver.NewDefaultAPIGroupInfo(imageapiv1.GroupName, c.Registry, c.Scheme, metav1.ParameterCodec, c.Codecs)
+	apiGroupInfo.GroupMeta.GroupVersion = imageapiv1.SchemeGroupVersion
+	apiGroupInfo.VersionedResourcesStorageMap[imageapiv1.SchemeGroupVersion.Version] = v1Storage
+	if err := s.GenericAPIServer.InstallAPIGroup(&apiGroupInfo); err != nil {
+		return nil, err
+	}
+
+	return s, nil
+}
+
+func (c *ImageAPIServerConfig) V1RESTStorage() (map[string]rest.Storage, error) {
+	c.makeV1Storage.Do(func() {
+		c.v1Storage, c.v1StorageErr = c.newV1RESTStorage()
+	})
+
+	return c.v1Storage, c.v1StorageErr
+}
+
+func (c *ImageAPIServerConfig) newV1RESTStorage() (map[string]rest.Storage, error) {
+	// TODO: allow the system CAs and the local CAs to be joined together.
+	importTransport, err := restclient.TransportFor(&restclient.Config{})
+	if err != nil {
+		return nil, fmt.Errorf("unable to configure a default transport for importing: %v", err)
+	}
+	insecureImportTransport, err := restclient.TransportFor(&restclient.Config{
+		TLSClientConfig: restclient.TLSClientConfig{
+			Insecure: true,
+		},
+	})
+	if err != nil {
+		return nil, fmt.Errorf("unable to configure a default transport for importing: %v", err)
+	}
+
+	deprecatedOpenshiftClientForImages, err := osclient.New(c.GenericConfig.LoopbackClientConfig)
+	if err != nil {
+		return nil, err
+	}
+	deprecatedOpenshiftClient, err := osclient.New(c.CoreAPIServerClientConfig)
+	if err != nil {
+		return nil, err
+	}
+	coreClient, err := coreclient.NewForConfig(c.CoreAPIServerClientConfig)
+	if err != nil {
+		return nil, err
+	}
+	authorizationClient, err := authorizationclient.NewForConfig(c.GenericConfig.LoopbackClientConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	imageStorage, err := imageetcd.NewREST(c.GenericConfig.RESTOptionsGetter)
+	if err != nil {
+		return nil, fmt.Errorf("error building REST storage: %v", err)
+	}
+	imageRegistry := image.NewRegistry(imageStorage)
+	imageSignatureStorage := imagesignature.NewREST(deprecatedOpenshiftClientForImages.Images())
```

Comment:
@soltysh we need to get rid of this

@deads2k can you make a TODO?

Reply:
> @deads2k can you make a TODO?

There are a few of these bouncing around in our separated servers.  I think we'll hunt them down based on package imports rather than todos.  

These are bugs after all, so they go in after freeze.  We're using the wrong API and need to switch to groupified.

Reply:
I'm currently working on having them removed from entire `pkg/image`.